### PR TITLE
Implement RecordEventsSpanImpl.toSpanProto(). Add more unit tests.

### LINKED ISSUE #279: SDK: Implement Span (SpanSdk) interface
The implementation of the Span should allow users to configure the maximum attributes, events, etc. to be kept and drop oldest data when these limits are hit.

Suggestion: Import the Span implementation from OpenCensus

### PR DESCRIPTION
**Pull request by songy23**
Fixes #279.

Tests are mostly copied from https://github.com/census-instrumentation/opencensus-java/blob/master/impl_core/src/test/java/io/opencensus/implcore/trace/RecordEventsSpanImplTest.java.

### CODE REVIEW

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsSpanImpl.java [hunk 1]
```diff
@@ -191,7 +226,7 @@ long getEndNanoTime() {
    */
   long getLatencyNs() {
     synchronized (this) {
-      return hasBeenEnded ? endNanoTime - startNanoTime : clock.nowNanos() - startNanoTime;
+      return getEndNanoTime() - startNanoTime;
```
Comment (bogdandrutu) [inline sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsSpanImpl.java]: Should we avoid double sync here? maybe have a private getEndNanoTimeUnlocked? Also for the moment this doesn't seem to be used, maybe add a test for this.

File: sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java [hunk 2]
```diff
@@ -51,6 +56,8 @@
 @RunWith(JUnit4.class)
 public class RecordEventsSpanImplTest {
   private static final String SPAN_NAME = "MySpanName";
+  private static final String SPAN_NEW_NAME = "NewName";
+  private static final long NANOS_PER_SECOND = (long) 1e9;
   private final TraceId traceId = TestUtils.generateRandomTraceId();
   private final SpanId spanId = TestUtils.generateRandomSpanId();
   private final SpanId parentSpanId = TestUtils.generateRandomSpanId();
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java]: Maybe a test for null parentSpanId?

File: sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsSpanImpl.java (hunk 1 above)
Comment (songy23) [reply sdk/src/main/java/io/opentelemetry/sdk/trace/RecordEventsSpanImpl.java]: Make sense, updated and added tests.

File: sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java (hunk 2 above)
Comment (songy23) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java]: Added in https://github.com/open-telemetry/opentelemetry-java/commit/b7e9be01903fcd5d912a0899c71e39186203e40e.

File: sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java [hunk 3]
```diff
@@ -59,6 +59,7 @@
   private static final String SPAN_NAME = "MySpanName";
   private static final String SPAN_NEW_NAME = "NewName";
   private static final long NANOS_PER_SECOND = (long) 1e9;
+  private static final long MILLIS_PER_SECOND = 1000;
```
Comment (bogdandrutu) [inline sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java]: TimeUnit.SECONDS.toMillis(1)? Same above.

Comment (songy23) [reply sdk/src/test/java/io/opentelemetry/sdk/trace/RecordEventsSpanImplTest.java]: SG, updated.

**Comment by bogdandrutu** [comment]
@songy23 please rebase :)

**Comment by songy23** [comment]
You caught me :) Rebased. Please help me merge this PR.

**Pull request merged**